### PR TITLE
Expose dynamo peak CUDA memory usage

### PR DESCRIPTION
**Pull request by huydhn**
As part of T169619666, I'm trying to expose the peak dynamo memory usage in GB as another dimension in the dashboard to supplement the current memory compression ratio metric (eager / dynamo).  This is the first line of defense if a change in PyTorch increases CUDA memory usage for more than x%

I only measure dynamo here because that's the focus of strengthening PyTorch reliability from my understanding, not eager.

### Testing

* Add a chart for peak CUDA memory usage in GB for dynamo https://torchci-git-fork-huydhn-expose-max-cuda-mem-588d7e-fbopensource.vercel.app/benchmark/compilers
* Add a column for peak CUDA memory usage in the detail page with 20% threshold https://torchci-git-fork-huydhn-expose-max-cuda-mem-588d7e-fbopensource.vercel.app/benchmark/torchbench/inductor_dynamic?startTime=Thu%2C%2030%20Nov%202023%2022%3A04%3A30%20GMT&stopTime=Thu%2C%2007%20Dec%202023%2022%3A04%3A30%20GMT&granularity=hour&mode=training&dtype=amp&lBranch=main&lCommit=622688fab9fc6d20ff3475a8a0a1fdb6af9d837e&rBranch=main&rCommit=7f4064034266d860a79fa7dbeaba30fde159b038 
* An example regression (moco/training/cudagraphs_dynamic/amp) https://torchci-git-fork-huydhn-expose-max-cuda-mem-588d7e-fbopensource.vercel.app/benchmark/torchbench/inductor_dynamic?startTime=Thu%2C%2030%20Nov%202023%2022%3A04%3A30%20GMT&stopTime=Thu%2C%2007%20Dec%202023%2022%3A04%3A30%20GMT&granularity=hour&mode=training&dtype=amp&lBranch=main&lCommit=622688fab9fc6d20ff3475a8a0a1fdb6af9d837e&rBranch=main&rCommit=729ac7317a50a6a195b324cf6cefd748bf4f5498&model=moco

### CODE REVIEW

File: torchci/pages/benchmark/[suite]/[compiler]/[[...page]].tsx [hunk 1]
```diff
@@ -611,6 +619,56 @@ function ModelPanel({
...
+                  }
+
+                  // Decreasing more than x%
+                  if (r - l > PEAK_MEMORY_USAGE_RELATIVE_THRESHOLD * r) {
+                    return styles.ok;
+                  }
+
+                  // Increasing more than x%
+                  if (l - r > PEAK_MEMORY_USAGE_RELATIVE_THRESHOLD * r) {
+                    return styles.error;
```
Comment (osalpekar) [inline torchci/pages/benchmark/[suite]/[compiler]/[[...page]].tsx:657]: How does this error reflect itself in the dashboard?

File: torchci/rockset/inductor/__sql/compilers_benchmark_performance.sql [hunk 2]
```diff
@@ -103,6 +105,14 @@ results AS (
       CAST(memory_bandwidth AS FLOAT),
       0.0
     ) AS memory_bandwidth,
+    IF(TRY_CAST(dynamo_peak_mem AS FLOAT) IS NOT NULL,
```
Comment (osalpekar) [inline torchci/rockset/inductor/__sql/compilers_benchmark_performance.sql:108]: IIUC this is a raw memory usage number. Do we want this as peak utilization (percentage of available memory) instead, since this is what is used for internal tracking? Will leave that up to whoever monitors this dashboard though

**Review by osalpekar — APPROVED** [review]
LGTM, cc @desertfire

File: torchci/pages/benchmark/[suite]/[compiler]/[[...page]].tsx (hunk 1 above)
Comment (huydhn) [reply torchci/pages/benchmark/[suite]/[compiler]/[[...page]].tsx:657]: Atm, the regression is highlight as a failure on the dashboard.  Here is an example for moco/training/cudagraphs_dynamic/amp https://torchci-git-fork-huydhn-expose-max-cuda-mem-588d7e-fbopensource.vercel.app/benchmark/torchbench/inductor_dynamic?startTime=Thu%2C%2030%20Nov%202023%2022%3A04%3A30%20GMT&stopTime=Thu%2C%2007%20Dec%202023%2022%3A04%3A30%20GMT&granularity=hour&mode=training&dtype=amp&lBranch=main&lCommit=622688fab9fc6d20ff3475a8a0a1fdb6af9d837e&rBranch=main&rCommit=729ac7317a50a6a195b324cf6cefd748bf4f5498&model=moco

File: torchci/pages/benchmark/[suite]/[compiler]/[[...page]].tsx [hunk 3]
```diff
@@ -68,10 +69,11 @@ const MIN_ENTRIES = 10;
...
 // Headers
 const ACCURACY_HEADER = "Accuracy";
-const SPEEDUP_HEADER = `Performance speedup (threshold = ${SPEEDUP_THRESHOLD}x)`;
-const ABS_LATENCY_HEADER = `Absolute execution time (millisecond)`;
+const SPEEDUP_HEADER = `Perf. speedup (threshold = ${SPEEDUP_THRESHOLD}x)`;
+const ABS_LATENCY_HEADER = `Abs. execution time (millisecond)`;
 const COMPILATION_LATENCY_HEADER = `Compilation latency (seconds)`;
-const MEMORY_HEADER = `Peak memory compression ratio (threshold = ${COMPRESSION_RATIO_THRESHOLD}x)`;
+const MEMORY_HEADER = `Peak mem compress ratio (threshold = ${COMPRESSION_RATIO_THRESHOLD}x)`;
+const PEAK_MEMORY_USAGE_HEADER = `Peak dynamo mem usage (threshold = ${PEAK_MEMORY_USAGE_RELATIVE_THRESHOLD}x)`;
```
Comment (desertfire) [inline torchci/pages/benchmark/[suite]/[compiler]/[[...page]].tsx:76]: Let's put `GB` in the title for clarification

**Review by desertfire — APPROVED** [review]
